synapseclient/core/credentials/credential_provider.py

```diff
@@ -104,47 +94,39 @@ def _create_synapse_credential(
 
 
 class UserArgsCredentialsProvider(SynapseCredentialsProvider):
+    def _get_auth_info(self, syn, user_login_args):
+
+        username = user_login_args.username
+        token = user_login_args.auth_token
+        # username = user_login_args.get("username")
+        # token = user_login_args.get("auth_token")
+
+        if username and token:
+            return username, token
+
+        return None, None
```

Comment:
It's valid to have a token without specifying your username.

We should also look to have types in place for the function arguments and return type:
`def _get_auth_info(self, syn, user_login_args):`